src/ecdh.rs

```diff
@@ -137,35 +109,17 @@ impl SharedSecret {
                 ptr::null_mut(),
             )
         };
-        debug_assert_eq!(res, 1); // The default `secp256k1_ecdh_hash_function_default` should always return 1.
+        // The default `secp256k1_ecdh_hash_function_default` should always return 1.
+        // and the scalar was verified to be valid(0 > scalar > group_order) via the type system
```

Comment:
"via the type system" confused me for a minute. Yes, it's a valid scalar, otherwise it wouldn't have type `SecretKey`. But our code ensures this, not the type system. (Totally okay to merge like this, just wanted to note this.)

Reply:
I think it's fair to say the "type system" is in charge of ensuring validity of data, if there is a type that the API prevents you from constructing in an invalid way.